source/Body.h

```diff
@@ -30,9 +31,24 @@ class Sprite;
 
 
 
+// Helper class to add a mutex to Body without declaring the full copy constructor.
+// These operations are not thread safe. Do not copy the object in a concurrent context,
+// or with a locked mutex.
+// The use of the mutex must be enforced externally.
+class WithMutex {
```

Comment:
> Each class gets its own header unless it is a nested class:Each class "MyClass" should have its own .h and .cpp files, and they should be named "MyClass.h" and "MyClass.cpp".
> 
> This is so that it is easy to find the implementation of any class you are interested in.

style guide